File: dotnet/RelevantCodes.ExtentReports/RelevantCodes.ExtentReports/Model/TestAttribute.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace RelevantCodes.ExtentReports.Model
{
    public abstract class TestAttribute
    {
        protected string _name;

        public string Name
        {
            get 
            {
                return _name;
            }
        }

        protected TestAttribute(string Name)
        {
            _name = Name;
        }
    }
}

File: src/main/java/com/aventstack/extentreports/model/TestAttribute.java
package com.aventstack.extentreports.model;

import java.io.Serializable;

public abstract class TestAttribute extends Attribute implements Serializable {

    static final long serialVersionUID = 1010210091204302766L;

    public TestAttribute(String k) {
        super(k);
    }
    
    public String getName() {
        return getKey();
    }
    
    public void setName(String name) {
        setKey(name);
    }
    
    public String getDescription() {
        return getValue();
    }
    
    public void setDescription(String description) {
        setValue(description);
    }
    
}